### PR TITLE
llvm build script

### PR DESCRIPTION
**Pull request by ChunyuLiao**

### CODE REVIEW

**Comment by qjivy** [comment]
Run the script on C7,  failed, output log:

```
Cloning into 'llvm-project'...
Updating files:   7% (9940/130374)^MUpdating files:   8% (10430/130374)^MUpdating files:   9% (11734/130374)^MUpdating files:  10% (13038/130374)^MUpdating files:  11% (14342/130374)^MUpdating files:  12% (15645/130374)^MUpdating files:  13% (16949/130374)^MUpdating files:  14% (18253/130374)^MUpdating files:  15% (19557/130374)^MUpdating files:  16% (20860/130374)^MUpdating files:  17% (22164/130374)^MUpdating files:  18% (23468/130374)^MUpdating files:  19% (24772/130374)^MUpdating files:  20% (26075/130374)^MUpdating files:  21% (27379/130374)^MUpdating files:  22% (28683/130374)^MUpdating files:  22% (28785/130374)^MUpdating files:  23% (29987/130374)^MUpdating files:  24% (31290/130374)^MUpdating files:  25% (32594/130374)^MUpdating files:  26% (33898/130374)^MUpdating files:  27% (35201/130374)^MUpdating files:  28% (36505/130374)^MUpdating files:  29% (37809/130374)^MUpdating files:  30% (39113/130374)^MUpdating files:  31% (40416/130374)^MUpdating files:  32% (41720/130374)^MUpdating files:  33% (43024/130374)^MUpdating files:  34% (44328/130374)^MUpdating files:  35% (45631/130374)^MUpdating files:  36% (46935/130374)^MUpdating files:  36% (47754/130374)^MUpdating files:  37% (48239/130374)^MUpdating files:  38% (49543/130374)^MUpdating files:  39% (50846/130374)^MUpdating files:  40% (52150/130374)^MUpdating files:  41% (53454/130374)^MUpdating files:  42% (54758/130374)^MUpdating files:  43% (56061/130374)^MUpdating files:  44% (57365/130374)^MUpdating files:  45% (58669/130374)^MUpdating files:  46% (59973/130374)^MUpdating files:  47% (61276/130374)^MUpdating files:  48% (62580/130374)^MUpdating files:  48% (62945/130374)^MUpdating files:  49% (63884/130374)^MUpdating files:  50% (65187/130374)^MUpdating files:  51% (66491/130374)^MUpdating files:  52% (67795/130374)^MUpdating files:  53% (69099/130374)^MUpdating files:  54% (70402/130374)^MUpdating files:  54% (70780/130374)^MUpdating files:  55% (71706/130374)^MUpdating files:  56% (73010/130374)^MUpdating files:  57% (74314/130374)^MUpdating files:  58% (75617/130374)^MUpdating files:  59% (76921/130374)^MUpdating files:  60% (78225/130374)^MUpdating files:  61% (79529/130374)^MUpdating files:  62% (80832/130374)^MUpdating files:  63% (82136/130374)^MUpdating files:  64% (83440/130374)^MUpdating files:  65% (84744/130374)^MUpdating files:  65% (85329/130374)^MUpdating files:  66% (86047/130374)^MUpdating files:  67% (87351/130374)^MUpdating files:  68% (88655/130374)^MUpdating files:  69% (89959/130374)^MUpdating files:  70% (91262/130374)^MUpdating files:  71% (92566/130374)^MUpdating files:  72% (93870/130374)^MUpdating files:  73% (95174/130374)^MUpdating files:  74% (96477/130374)^MUpdating files:  75% (97781/130374)^MUpdating files:  75% (98355/130374)^MUpdating files:  76% (99085/130374)^MUpdating files:  77% (100388/130374)^MUpdating files:  78% (101692/130374)^MUpdating files:  79% (102996/130374)^MUpdating files:  80% (104300/130374)^MUpdating files:  81% (105603/130374)^MUpdating files:  82% (106907/130374)^MUpdating files:  83% (108211/130374)^MUpdating files:  84% (109515/130374)^MUpdating files:  85% (110818/130374)^MUpdating files:  86% (112122/130374)^MUpdating files:  87% (113426/130374)^MUpdating files:  88% (114730/130374)^MUpdating files:  89% (116033/130374)^MUpdating files:  89% (116211/130374)^MUpdating files:  90% (117337/130374)^MUpdating files:  91% (118641/130374)^MUpdating files:  92% (119945/130374)^MUpdating files:  93% (121248/130374)^MUpdating files:  94% (122552/130374)^MUpdating files:  95% (123856/130374)^MUpdating files:  96% (125160/130374)^MUpdating files:  97% (126463/130374)^MUpdating files:  98% (127767/130374)^MUpdating files:  99% (129071/130374)^MUpdating files: 100% (130374/130374)^MUpdating files: 100% (130374/130374), done.
Note: switching to 'remotes/origin/release/15.x'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may 
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 8dfdcc7b7bf6 [libc++] Fix memory leaks when throwing inside std::vector constructors
permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/create?fromImage=openeuler%2Fopeneuler&tag=23.03": dial unix /var/run/docker.sock: connect: permission denied
docker: permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/create?name=llvm-chunyu": dial unix /var/run/docker.sock: connect: permission denied. 
See 'docker run --help'.
permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/llvm-chunyu/json": dial unix /var/run/docker.sock: connect: permission denied
cp: cannot stat 'rvoe23_03_llvm.tar.gz': No such file or directory
permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/llvm-chunyu/json": dial unix /var/run/docker.sock: connect: permission denied
permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/llvm-chunyu/stop": dial unix /var/run/docker.sock: connect: permission denied
```

File: run-llvm.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+cd llvm-project
+git checkout remotes/origin/release/15.x
+docker pull openeuler/openeuler:23.03
+docker run -itd --restart=always -v  $PWD:/home/llvm-project --name llvm-chunyu openeuler/openeuler:23.03 /bin/bash
+docker exec llvm-chunyu /bin/sh -c "dnf install -y cmake ninja-build autoconf automake make python3 libmpc-devel mpfr-devel gmp-devel gawk  bison flex texinfo patchutils gcc gcc-c++ zlib-devel expat-devel diffutils;
+cd /home/llvm-project && mkdir rvoe23_03_llvm && cd rvoe23_03_llvm;
+cmake -DLLVM_ENABLE_PROJECTS="clang" -DCMAKE_BUILD_TYPE="Release"  -G Ninja ../llvm;
+ninja;
+cd ../;
+tar -czvf rvoe23_03_llvm.tar.gz rvoe23_03_llvm/bin/;"
```
Comment (Xeonacid) [inline run-llvm.sh]: 只打包bin恐怕不能工作

Comment (Xeonacid) [inline run-llvm.sh]:
```suggestion
docker exec llvm-chunyu /bin/sh -c "dnf install -y cmake ninja-build autoconf automake make python3 libmpc-devel mpfr-devel gmp-devel gawk  bison flex texinfo patchutils gcc gcc-c++ zlib-devel expat-devel diffutils &&
cd /home/llvm-project && mkdir rvoe23_03_llvm && cd rvoe23_03_llvm &&
cmake -DLLVM_ENABLE_PROJECTS="clang" -DCMAKE_BUILD_TYPE="Release"  -G Ninja ../llvm &&
ninja &&
cd ../ &&
tar -czvf rvoe23_03_llvm.tar.gz rvoe23_03_llvm/bin/;"
```

如果前面出错了就不执行后面的工作了
这个sh本身也改成这样会更好

File: run-llvm.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+git clone git@github.com:llvm/llvm-project.git
+cd llvm-project
+git checkout remotes/origin/release/15.x
```
Comment (Xeonacid) [inline run-llvm.sh]:
```suggestion
git clone https://github.com/llvm/llvm-project.git --depth 1 --branch release/15.x
cd llvm-project
```
- 使用https，不需要SSH key
- 指定depth 1和branch，减少下载量

File: run-llvm.sh (hunk 1 above)
Comment (ChunyuLiao) [reply run-llvm.sh]: 那是打rvoe23_03_llvm - bin的上层目录，还是把llvm-project源代码也加上呢？

Comment (ChunyuLiao) [reply run-llvm.sh]: 现在是打包了rvoe23_03_llvm

**Comment by ChunyuLiao** [comment]
updated:
1.  fix git checkout error.  (git clone https://github.com/llvm/llvm-project.git --depth 1 --branch release/15.x)
2. address comment from @Xeonacid.
thanks

**Comment by xen0n** [comment]
The current packaging infra that targets each distro individually is being replaced, so closing for now.

**Comment by imkiva** [comment]
@xen0n Hi, is there any link to the new "the packaging infra" replacing this repo? Wondering if we could start porting this script to it.

**Comment by xen0n** [comment]
@imkiva I've just pushed the current set of scripts I've been using; the `main` branch is now replaced and ready for refactoring. For now it's basically just simple containerized packaging using a greatest-common-denominator distro (Ubuntu 20.04) and cross-compiling to the alt-arches (aarch64 and riscv64), but I hope to gradually morph the scripts into something more amenable to automation *and* more maintainable than the previous ad-hoc set of distro-specific scripts.

**Pull request closed without merging**